superset/models/core.py

```diff
@@ -635,15 +635,25 @@ def get_all_schema_names(  # pylint: disable=unused-argument
 
     @property
     def db_engine_spec(self) -> Type[db_engine_specs.BaseEngineSpec]:
-        return self.get_db_engine_spec_for_backend(self.backend)
+        sqlalchemy_url = make_url_safe(self.sqlalchemy_uri_decrypted)
+        backend = sqlalchemy_url.get_backend_name()
+        driver = sqlalchemy_url.get_driver_name()
+
+        return self.get_db_engine_spec(backend, driver)
 
     @classmethod
     @memoized
-    def get_db_engine_spec_for_backend(
-        cls, backend: str
+    def get_db_engine_spec(
+        cls,
+        backend: str,
+        driver: str,
     ) -> Type[db_engine_specs.BaseEngineSpec]:
         engines = db_engine_specs.get_engine_specs()
-        return engines.get(backend, db_engine_specs.BaseEngineSpec)
+        for engine in engines.values():
+            if engine.engine == backend and engine.driver == driver:
+                return engine
+
+        raise Exception("No engine found")
```

Comment:
For backward compatibility of connections that aren't using a "supported" driver, should we default to just a matching engine name if no engine+driver combo is found instead of raising an error?

Reply:
Yeah, that's a great idea!